dataset_loaders/videos/davis2017.py

```diff
@@ -0,0 +1,269 @@
+                 split=.75,
+                 *args, **kwargs):
+
+        self.which_set = which_set
+        self.threshold_masks = threshold_masks
+        self.multiobject = multiobject
+        if dataset_version not in ['2016', '2017']:
+            raise RuntimeError('Unknown dataset version')
+        self._image_sets_path = os.path.join(self.path, 'ImageSets',
+                                             dataset_version)
+
+        self.image_path = os.path.join(self.path,
+                                       'JPEGImages', '480p')
+        self.mask_path = os.path.join(self.path,
+                                      'Annotations', '480p')
+
+        # Prepare data paths
+        if 'train' in self.which_set or 'val' in self.which_set:
+            self.split = split
+        elif 'test-dev' in self.which_set:
+            self.split = 1.
+        else:
+            raise RuntimeError('Unknown set')
+
+        if 'test-dev' in self.which_set and not self.multiobject:
+            raise RuntimeError('Single object instance is not '
+                               'available for test-dev in Davis2017')
+
+        super(Davis2017Dataset, self).__init__(*args, **kwargs)
+
+    def get_names(self):
+        """Return a dict of names, per prefix/subset."""
+        per_video_names = {}
+
+        # Populate self.filenames and self.prefix_list
+        filenames = self.filenames
+        prefix_list = self.prefix_list
+
+        # cycle through the different videos
+        for prefix in prefix_list:
+            exp_prefix = prefix + '/'
+            per_video_names[prefix] = [el.lstrip(exp_prefix) for el in
+                                       filenames if el.startswith(exp_prefix)]
+        return per_video_names
+
+    def load_sequence(self, sequence):
+        """Load a sequence of images/frames
+
+        Auxiliary function that loads a sequence of frames with
+        the corresponding ground truth and their filenames.
+        Returns a dict with the images in [0, 1], their corresponding
+        labels, their subset (i.e. category, clip, prefix) and their
+        filenames.
+        """
+        from skimage import io
+
+        X = []
+        Y = []
+        F = []
+
+        rgbs = self.unique_rgbs
+
+        for prefix, frame_name in sequence:
+            frame = prefix + '/' + frame_name
+
+            img = io.imread(os.path.join(self.image_path, frame + 'jpg'))
+            img = img.astype(floatX) / 255.
+
+            if self.which_set == 'train' or self.which_set == 'valid':
+                mask = cv2.imread(os.path.join(self.mask_path, frame + 'png'))
+                # Convert mask from RGB to ids format
+                _id = 0
+                temp_mask = np.zeros(mask.shape[:-1], np.int32)
+                for rgb in sorted(rgbs[prefix]):
+                    temp_mask[np.where(np.all(mask == rgb,
+                                              -1).astype(int) == 1)] = _id
```

Comment:
Why do you need a temporary mask? Also, this code is unreadable. This should be equivalent, please check it out:
`mask[mask == rgb)] = _id`

You could also use a dictionary as done [here](https://github.com/fvisin/dataset_loaders/blob/29451cbdc8f3b5606d30cf80349c4f5beae59c12/dataset_loaders/parallel_loader.py#L636-L644).

Reply:
Ok I remember why a temporary mask was needed. The labels read from file (mask variable) has 3-channel depth and I cannot modify it since it is useful to compare the RGB values. So in order to convert the RGBs in ids I had to initialize a new zero array with one channel depth to be filled in the for loop.

Reply:
Would 
```python 
for rgb in sorted(rgbs[prefix]):
    mask[mask == rgb)][0] = _id
mask = mask[0]
```
work?

Reply:
No, mask = np.array(Image.open('path')) makes mask variable read-only. So I cannot perform any assignment. I can solve by create a local copy by np.copy but is the same thing to create a temporary mask..

Reply:
Are you sure that you are doing `mask = np.array(Image.open('path'))` and not `mask = np.asarray(Image.open('path'))`? I thought the former to create a new and editable np.array. If that's not the case proceed with the temp array.

Reply:
This works inplace:
```python
In [1]: from PIL import Image

In [2]: img = Image.open('test.jpg')

In [3]: import numpy as np

In [4]: npimg = np.array(img)

In [5]: npimg.shape
Out[5]: (960, 487, 3)

In [6]: npimg[0, 0]
Out[6]: array([182, 222, 221], dtype=uint8)

In [7]: npimg[0, 0] = [0,0,0]

In [8]: npimg[0, 0]
Out[8]: array([0, 0, 0], dtype=uint8)
```

Reply:
Yes but I have to fill the mask with ids not with RGB values.